src/services/post.service.ts

```diff
@@ -0,0 +1,112 @@
+import React from 'react';
+import { Action, ActionTypes, NotificationTypes } from '../reducer/store';
+import { getPostComments, getUserPost } from '../api/users.api';
+import { User } from '../types/User';
+import { Post } from '../types/Post';
+
+export function postService(
```

Comment:
This service file mixes the service layer and the UI layer by using React.Dispatch and React.SetStateAction. It would be better to handle UI state changes in the components themselves and keep the service layer only for data fetching and data manipulation.